pkg/kubectl/cmd/delete.go

```diff
@@ -148,7 +163,10 @@ func RunDelete(f cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []stri
 		if gracePeriod != -1 {
 			return fmt.Errorf("--now and --grace-period cannot be specified together")
 		}
-		gracePeriod = 0
+		gracePeriod = 1
+	}
+	if gracePeriod == 0 && !cmdutil.GetFlagBool(cmd, "force") {
+		return fmt.Errorf("Immediate deletion may result in multiple copies of your resources running at the same time. You must pass --force to delete with grace period 0.")
```

Comment:
I think this error message may not be clear to someone who isn't familiar with the entire issue. Also, if it's force deleting a simple pod object, the multiple copies comment does not apply. Maybe we could rephrase it to say something along the lines of:

> Immediate deletion does not wait for confirmation that the running resource has been terminated and may leave it running in case of a network partition.